src/v/kafka/server/group_manager.cc

```diff
@@ -106,10 +106,17 @@ void group_manager::attach_partition(ss::lw_shared_ptr<cluster::partition> p) {
 
 ss::future<> group_manager::cleanup_removed_topic_partitions(
   const std::vector<model::topic_partition>& tps) {
-    using group_info = absl::node_hash_map<group_id, group_ptr>::value_type;
-    return ss::do_for_each(_groups, [this, &tps](group_info& gi) {
-        return gi.second->remove_topic_partitions(tps).then(
-          [this, g = gi.second] {
+    // operate on a light-weight copy of group pointers to avoid iterating over
+    // the main index which is subject to concurrent modification.
+    std::vector<group_ptr> groups;
```

Comment:
use `std::copy()` it automagically `.reserves()` the sizze of the `std::distance(begin,end)` for you. 

you might have to use a `std::transform` 

in any case, just make sure you do `groups.serve(_groups.size())`

Reply:
oh sweet didn't realize std::copy did that